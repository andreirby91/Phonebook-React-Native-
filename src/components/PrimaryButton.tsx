import React from 'react';
import {TouchableOpacity, Text, StyleSheet, ViewStyle} from 'react-native';

type Props = {
  onPress: () => void;
  title: string;
  customStyle?: ViewStyle;
};

const styles = StyleSheet.create({
  button: {
    height: 50,
    minWidth: 200,
    backgroundColor: '#0984e3',
    borderRadius: 30,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
  },
});

const PrimaryButton: React.FC<Props> = ({onPress, title, customStyle}) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, customStyle]}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );
};

export default PrimaryButton;
