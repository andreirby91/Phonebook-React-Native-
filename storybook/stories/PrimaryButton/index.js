import React from 'react';
import PrimaryButton from '../../../src/components/PrimaryButton';

const PrimaryButtonStory = () => {
  return (
    <PrimaryButton
      title="Primary Button"
      onPress={() => alert('Button pressed!')}
    />
  );
};

export default PrimaryButtonStory;
