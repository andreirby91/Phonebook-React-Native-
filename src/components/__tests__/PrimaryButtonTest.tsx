import React from 'react';
import renderer from 'react-test-renderer';

import PrimaryButton from '../PrimaryButton';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<PrimaryButton title="Button" onPress={() => {}} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
