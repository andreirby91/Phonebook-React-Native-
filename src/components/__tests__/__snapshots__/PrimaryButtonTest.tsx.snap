// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0984e3",
      "borderRadius": 30,
      "height": 50,
      "justifyContent": "center",
      "minWidth": 200,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
      }
    }
  >
    Button
  </Text>
</View>
`;
